api/models/model.py

```diff
@@ -294,6 +294,10 @@ def tags(self):
 
         return tags or []
 
+    @property
+    def mcp_server(self):
+        return db.session.query(AppMCPServer).filter(AppMCPServer.app_id == self.id).first()
```

Comment:
Avoid placing database queries (or any IO operations) inside property getters/setters. This pattern obscures the fact that property access triggers IO operations and can easily lead to N+1 query problems, as demonstrated in the following example:

```python
for app in apps:
    mcp_server = app.mcp_server  # Each access triggers a database query
```

In this loop, each `mcp_server` property access issues a separate database query, resulting in N queries total instead of a single optimized query.